### PR TITLE
Add Next and Back Navigation Buttons to the Documentation

### PR DESCRIPTION
**Pull request by corwinjoy**
Add navigation buttons to the documentation as per the proposal here: https://github.com/duckdb/duckdb-web/issues/2852

Comments:
1. I was able to get this to work pretty well but there were a couple limitations.
   a. The Sitemap appears twice, once at the top level and once at the beginning of the documentation. I had to pick one to get a single navigation order so I chose the Sitemap position in the left menu.
   b. I did not include navigation for the demo page since I think that makes no sense.

2. I made the buttons as pretty as I know how, but I'm not all that great at graphic design. The buttons are a class in the css file so I welcome suggestions there.

3. It might make sense to have navigation buttons at the top of the page as well, but I am not as sure how, or if, I should do that.

Thanks in advance for taking a look! I do think this does help to improve the docs.

### CODE REVIEW

**Comment by corwinjoy** [comment]
I did also run the linter as suggested by the contribution guide. I didn't see any errors from this addition, but it did come back pretty dirty / suggesting changes for the existing code so I'm not sure what to make of that.

File: _data/menu_docs_dev.json [hunk 1]
```diff
@@ -9,10 +9,6 @@
       "page": "Documentation",
       "slug": "",
       "mainfolderitems": [
-        {
-          "page": "Overview",
-          "url": "index"
-        },
```
Comment (corwinjoy) [inline _data/menu_docs_dev.json:15]: As mentioned, this page is pointed to both the the docs and the sitemap link. I had to choose a single location in order to get a unique navigation order.

**Comment by szarnyasg** [comment]
Hi, thanks for this PR! I passed it along to our design team for feedback. Will let you know the outcome.

**Comment by corwinjoy** [comment]
@szarnyasg Very cool! Does the design team have any questions or feedback? I think this would be a helpful feature for navigating the web documentation.

**Comment by corwinjoy** [comment]
Hey @szarnyasg great meeting you at the con today and it was nice to chat! I thought I might follow up on this idea since I never did hear anything back...

**Comment by szarnyasg** [comment]
Hi @corwinjoy I designed it with our designers and it's difficult – both design-wise but also logic-wise, it's not always obvious where it would lead. So we decided to postpone the implementation of this to the future. Sorry about that.

**Comment by corwinjoy** [comment]
No problem, thank you for letting me know!

**Pull request closed without merging**